### PR TITLE
nil_only_functions fix: Don't trigger when nil and non nil are returned.

### PR DESCRIPTION
**Pull request by rstorr**

### CODE REVIEW

File: tenets/codelingo/go/nil_only_functions/codelingo.yaml [hunk 1]
```diff
@@ -1,3 +1,10 @@
+functions:
```
Comment (waigani) [inline tenets/codelingo/go/nil_only_functions/codelingo.yaml]: new syntax is `funcs`

File: tenets/codelingo/go/nil_only_functions/expected.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
+[
+  {
+   "name": "nil-only-functions",
```
Comment (waigani) [inline tenets/codelingo/go/nil_only_functions/expected.json]: Are you using the latest review flow? The result format should be different.

File: tenets/codelingo/go/nil_only_functions/codelingo.yaml (hunk 1 above)
Comment (rstorr) [reply tenets/codelingo/go/nil_only_functions/codelingo.yaml]: keep as functions until next deploy..

File: tenets/codelingo/go/nil_only_functions/expected.json (hunk 2 above)
Comment (rstorr) [reply tenets/codelingo/go/nil_only_functions/expected.json]: will update.

**Pull request merged**